src/source/src/parser/canal/simd_json_parser.rs

```diff
@@ -79,10 +80,15 @@ impl CanalJsonParser {
                 let results = inserted
                     .into_iter()
                     .map(|v| {
+                        let field_mapping = get_keys_from_value(v);
                         writer.insert(|column| {
                             cannal_simd_json_parse_value(
                                 &column.data_type,
-                                v.get(column.name.as_str()),
+                                get_column_from_value(
+                                    column.name.to_lowercase().as_str(),
```

Comment:
How about passing the original value and using `eq_ignore_ascii_case` to reduce an allocation? I guess the optimization makes sense here because it's likely in the critical path.

Reply:
Well, to validate these choices, perhaps we should run the microbenchmark? https://github.com/risingwavelabs/risingwave/blob/main/src/source/benches/json_parser.rs